### PR TITLE
Fix crash during loading

### PR DESCRIPTION
**Pull request by Arias800**
The launcher was crashing during the loading of the channel. On my side, the culprit was "com.android.video".
But it may happen with some others applications too.

### CODE REVIEW

**Comment by nielsvanvelzen** [comment]
Thanks for the PR, can you share the exception you got before this fix?

**Comment by Arias800** [comment]
The error before my fix was this one : 

```
-10-16 08:17:39.282 23459-23459 AndroidRuntime          nl.ndat.tvlauncher                   E  FATAL EXCEPTION: main
                                                                                                    Process: nl.ndat.tvlauncher, PID: 23459
                                                                                                    java.lang.IllegalStateException: Need app link intent uri for channel. Use method setAppLinkIntent or setAppLinkIntentUri to set it.
                                                                                                    	at androidx.tvprovider.media.tv.PreviewChannel$Builder.build(PreviewChannel.java:566)
                                                                                                    	at androidx.tvprovider.media.tv.PreviewChannel.fromCursor(PreviewChannel.java:107)
                                                                                                    	at nl.ndat.tvlauncher.data.resolver.ChannelResolver.getPreviewChannels(ChannelResolver.kt:43)
                                                                                                    	at nl.ndat.tvlauncher.data.repository.ChannelRepository.refreshPreviewChannels(ChannelRepository.kt:62)
                                                                                                    	at nl.ndat.tvlauncher.data.repository.ChannelRepository.refreshAllChannels(ChannelRepository.kt:58)
                                                                                                    	at nl.ndat.tvlauncher.data.repository.ChannelRepository$refreshAllChannels$1.invokeSuspend(Unknown Source:14)
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
                                                                                                    	at androidx.lifecycle.DispatchQueue.drainQueue(DispatchQueue.kt:75)
                                                                                                    	at androidx.lifecycle.DispatchQueue.enqueue(DispatchQueue.kt:112)
                                                                                                    	at androidx.lifecycle.DispatchQueue.dispatchAndEnqueue$lambda-2$lambda-1(DispatchQueue.kt:100)
                                                                                                    	at androidx.lifecycle.DispatchQueue.$r8$lambda$G2ay370n_s_ksSHUJaD9zIU8eCw(Unknown Source:0)
                                                                                                    	at androidx.lifecycle.DispatchQueue$$ExternalSyntheticLambda0.run(Unknown Source:4)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:938)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loop(Looper.java:223)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7668)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
                                                                                                    	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@12b48db, Dispatchers.Main.immediate]
2022-10-16 08:17:39.290 23459-23459 Process                 nl.ndat.tvlauncher                   I  Sending signal. PID: 23459 SIG: 9
---------------------------- PROCESS ENDED (23459) for package nl.ndat.tvlauncher ----------------------------
2022-10-16 08:17:39.330  3704-3769  InputDispatcher         pid-3704                             E  channel '61212be nl.ndat.tvlauncher/nl.ndat.tvlauncher.LauncherActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
```

**Comment by nielsvanvelzen** [comment]
This is the check in the PreviewChannel.java source:
```
if (TextUtils.isEmpty(mValues.getAsString(Channels.COLUMN_APP_LINK_INTENT_URI))) {
  throw new IllegalStateException("Need app link intent uri for channel."
     + " Use method setAppLinkIntent or setAppLinkIntentUri to set it.");
}
```

So I guess it might be better to check if that specific column is null or empty.

**Comment by Arias800** [comment]
You're right, it's better to do this way.
Ignore the second commit, I just made a mistake.

File: app/src/main/kotlin/nl/ndat/tvlauncher/data/resolver/ChannelResolver.kt [hunk 1]
```diff
@@ -39,9 +39,12 @@ class ChannelResolver {
 			do {
 				try {
 					val packageName = cursor.getString(PreviewChannel.Columns.COL_PACKAGE_NAME)
-					if (packageName != "com.google.android.tvrecommendations") add(
-						PreviewChannel.fromCursor(cursor).toChannel()
-					)
+					// Check if app has channel.
+					if (!cursor.getString(PreviewChannel.Columns.COL_APP_LINK_INTENT_URI).isNullOrEmpty()){
+						if (packageName != "com.google.android.tvrecommendations") add(
```
Comment (nielsvanvelzen) [inline app/src/main/kotlin/nl/ndat/tvlauncher/data/resolver/ChannelResolver.kt:44]: If you merge these two if statements it should be good to merge 👍

Comment (Arias800) [reply app/src/main/kotlin/nl/ndat/tvlauncher/data/resolver/ChannelResolver.kt]: Thanks for the advice.
I have merged these if statement.

**Review by nielsvanvelzen — APPROVED** [review]

**Comment by nielsvanvelzen** [comment]
Thanks for your contribution!

**Pull request merged**